tower/tests/builder.rs

```diff
@@ -22,6 +22,7 @@ async fn builder_service() {
         .layer(RateLimitLayer::new(5, Duration::from_secs(1)))
         .layer(RetryLayer::new(policy))
         .layer(BufferLayer::new(5))
+        .map(|r: &'static str| r.to_string())
```

Comment:
I don't think this actually tests that this call changes the type. Would be good to have that in there somewhere?